libs/s3/objects.go

```diff
@@ -0,0 +1,70 @@
+package s3
+
+import (
+	"io/ioutil"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// GetObject returns an object from S3
+func (s S3Client) GetObject(key string) ([]byte, error) {
+	input := &s3.GetObjectInput{
+		Bucket: aws.String(s.Bucket),
+		Key:    aws.String(key),
+	}
+
+	result, err := s.Service.GetObject(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case s3.ErrCodeNoSuchKey:
+				log.Println(s3.ErrCodeNoSuchKey, aerr.Error(), input)
+				return []byte{}, err
+			default:
+				log.Println(aerr.Error(), input)
+			}
+		} else {
+			log.Println(err.Error(), input)
+		}
+		return nil, err
+	}
```

Comment:
This is a little convoluted, maybe something more like

```golang
if err != nil {
    log.Println(err.Error(), input)
    if aerr, ok := err.(awserr.Error); ok && aerr == s3.ErrCodeNoSuchKey {
        return []byte{}, err
    }
    return nil, err
}
```

Reply:
💯